kernel/src/main/java/org/kframework/compile/ExpandGroupAttribute.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) K Team. All Rights Reserved.
+package org.kframework.compile;
+
+import org.kframework.attributes.Att;
+import org.kframework.attributes.HasLocation;
+import org.kframework.kil.Definition;
+import org.kframework.kil.Module;
+import org.kframework.kil.Syntax;
+import org.kframework.utils.errorsystem.KEMException;
+
+/**
+ * Replace every attribute [group(att1,...,attN)] with the underlying attributes [att1,...,attN]
+ */
+public class ExpandGroupAttribute {
+
+    private static Att expandGroup(Att att, HasLocation node) {
+        if (!att.contains(Att.GROUP(), String.class)) {
+            return att;
+        }
+        String groups = att.get(Att.GROUP()).trim();
+        if (groups.isEmpty()) {
+            throw KEMException.compilerError("group(_) attribute expects a comma-separated list of arguments.");
+        }
+        KEMException badCommaException =
+                KEMException.compilerError("Extraneous ',' in group(_) attribute.", node);
+        if (groups.startsWith(",") || groups.endsWith(",")) {
+            throw badCommaException;
+        }
+        for (String group : groups.split("\\s*,\\s*")) {
+            if (group.isEmpty()) {
+                throw badCommaException;
+            }
+            if (Att.getWhitelistedOptional(group).isPresent()) {
+                throw KEMException.compilerError("User-defined group '" + group +
+                        "' conflicts with built-in attribute.");
+            }
+            if (!group.matches("[a-z][a-zA-Z0-9-]*")) {
+                throw KEMException.compilerError("Invalid argument '" + group + "' in group(_) attribute. " +
+                        "Expected a lower case letter followed by any number of alphanumeric or '-' characters.", node);
+            }
+            att = att.addGroup(group);
+        }
+        return att.remove(Att.GROUP());
```

Comment:
This looks a bit overkill.
A simple split and trim should be enough (like [here](https://github.com/runtimeverification/k/blob/master/kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java#L1868))
No need to change it though.